bridges/AmazonPriceTrackerBridge.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+class AmazonPriceTrackerBridge extends BridgeAbstract {
+
+	const MAINTAINER = 'captn3m0';
+	const NAME = 'Amazon Price Tracker';
+	const URI = 'https://www.amazon.com/';
+	const CACHE_TIMEOUT = 3600; // 1h
+	const DESCRIPTION = 'Tracks price for a single product on Amazon';
+
+	const PARAMETERS = [
+		[
+		'asin' => [
+			'name' => 'ASIN',
+			'required' => true,
+		],
+		'tld' => [
+			'name' => 'Country',
+			'type' => 'list',
+			'required' => true,
+			'values' => [
+				'Australia' => 'com.au',
+				'Brazil' => 'com.br',
+				'Canada' => 'ca',
+				'China' => 'cn',
+				'France' => 'fr',
+				'Germany' => 'de',
+				'India' => 'in',
+				'Italy' => 'it',
+				'Japan' => 'co.jp',
+				'Mexico' => 'com.mx',
+				'Netherlands' => 'nl',
+				'Spain' => 'es',
+				'United Kingdom' => 'co.uk',
+				'United States' => 'com',
+			],
+			'defaultValue' => 'com',
+		],
+	]];
+
+	public function getName(){
+		if(!is_null($this->getInput('tld')) && !is_null($this->getInput('q'))) {
+			return 'Amazon.'.$this->getInput('tld').': '.$this->getInput('q');
+		}
+
+		return parent::getName();
+	}
+
+	private function makeDomainName() {
+		return 'https://www.amazon.' . $this->getInput('tld');
+	}
+
+	private function makeUri() {
+		return $this->makeDomainName() . '/dp/' . $this->getInput('asin') . '/';
+	}
+
+	public function collectData() {
+		$uri = $this->makeUri();
+
+		$html = getSimpleHTMLDOM($uri)
+			or returnServerError('Could not request Amazon.');
+
+		// <img class="... frontImage" data-a-dynamic-image="{&quot;<image_url>&quot;:[$width,$height]}">
+		$imageHtml = $html->find('.frontImage')[0]->getAttribute('data-a-dynamic-image');
```

Comment:
Not sure why, but my HTML looks very different from what you expect. There is no `.frontImage` in my source, so this line fails on me, using this query: `?action=display&bridge=AmazonPriceTracker&asin=B071GB1VMQ&tld=com&format=Html`

Reply:
Went through the html and found an interesting element that we can use:

`<div id="cerberus-data-metrics" style="display: none;" data-asin="B00WTHJ5SU" data-asin-price="14.99" data-asin-shipping="0" data-asin-currency-code="USD" data-substitute-count="-1" ...>`